doc/proposals/helm-operator.md

```diff
@@ -0,0 +1,137 @@
+## Helm Operator Proposal
+
+### Background
+
+As was mentioned in the [Ansible Operator Proposal](./ansible-operator.md), not everyone is a golang developer, so the SDK needs to support other types of operators to gain adoption across a wider community of users.
+
+[Helm](https://helm.sh/) is one of the most widely-used tools for Kubernetes application management, and it bills itself as the "package manager for Kubernetes." Operators serve a nearly identical function, but they improve on Helm's concepts by incorporating an always-on reconciliation loop rather than relying on an imperative user-driven command line tool. By integrating Helm's templating engine and release management into an operator, the SDK will further increase the number of potential users by adding the ability to deploy Charts (e.g. from Helm's [large catalog of existing Charts](https://github.com/helm/charts)) as operators with very little extra effort.
+
+### Goals
+
+The goal of the Helm Operator will be to create a fully functional framework for Helm Chart developers to create operators. It will also expose a library for golang users to use Helm in their operator if they so choose. These two goals in conjunction will allow users to select the best technology for their project or skillset.
+
+### New Operator Type
+
+This proposal creates a new type of operator called `helm`. The new type is used to tell the tooling to act on that type of operator.
+
+### Package Structure
+
+Packages will be added to the operator-sdk. These packages are designed to be usable by the end user if they choose to and should have a well documented public API. The proposed packages are:
+
+* /operator-sdk/pkg/helm/client
+  * Will contain a helper function to create a Helm client from `controller-runtime` manager.
+
+* /operator-sdk/pkg/helm/controller
+  * Will contain the Helm controller.
+  * Will contain an exposed reconciler. The default `Add` method will use this reconciler.
+
+* /operator-sdk/pkg/helm/engine
+  * Will contain a Helm Engine implementation that adds owner references to generated Kubernetes resource assets, which is necessary for garbage collection of Helm chart resources.
+
+* /operator-sdk/pkg/helm/internal
+  * Will contain types and utilities used by other Helm packages in the SDK.
+
+* /operator-sdk/pkg/helm/release
+  * Will contain the Manager types and interfaces. A Manager is responsible for:
```

Comment:
@joelanford Yeah `Manager` would be a little confusing with the controller-runtime's Manager.
How about `release.ReleaseManager`? That's a little more descriptive than Manager and distinguishes it from the runtime's Manager.

Reply:
>release.ReleaseManager

That stutters I think it should be `release.Manager`. The namespace clearly delineates it from the controller-runtime `manager.Manager` IMO. 

> Avoid stutter. Since client code uses the package name as a prefix when referring to the package contents, the names for those contents need not repeat the package name. The HTTP server provided by the http package is called Server, not HTTPServer. Client code refers to this type as http.Server, so there is no ambiguity.

- https://blog.golang.org/package-names

Reply:
Good point @shawn-hurley. That was my initial instinct as well. My main concern was that there will be function calls and variables names for both types of Managers in Helm operator's `main.go`. 

Thinking about this more, I think `release.Manager` probably makes the most sense, and we can make sure the `main.go` uses variable names like `releaseManager` and `crManager` where appropriate to minimize confusion.

@hasbro17 Thoughts?